examples/example_parametrized_nodes.py

```diff
@@ -0,0 +1,160 @@
+"""
+Example for evolutionary regression with parametrized nodes
+===========================================================
+
+Example demonstrating the use of Cartesian genetic programming for a
+regression task that requires fine tuning of constants in parametrized
+nodes. This is achieved by introducing a new node, "ParametrizedAdd"
+which produces a scaled and shifted version of the sum of its inputs.
+
+"""
+
+import functools
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.constants
+import torch
+
+import cgp
+
+
+# %%
+# We first define a new node that adds two inputs then scales and
+# finally shifts the result. The scale ("w") and shift factors ("b")
+# are parameters that are adapted by local search. We need to define
+# the arity of the node, callables for the initial values for the
+# parameters and the operation of the node as a string. In this string
+# parameters are enclosed in angle brackets, inputs are denoted by "x_i"
+# with i representing their corresponding index.
+
+
+class ParametrizedAdd(cgp.OperatorNode):
+    """A node that adds its two inputs.
+
+    The result of addition is scaled by w and shifted by b. Both these
+    parameters can be adapted via local search are passed on from
+    parents to their offspring.
+
+    """
+
+    _arity = 2
+    _initial_values = {"<w>": lambda: 1.0, "<b>": lambda: 0.0}
+    _def_output = "<w> * (x_0 + x_1) + <b>"
+
+
+# %%
+# We define a target function which contains numerical constants that
+# are not available as constants for the search and need to be found
+# by local search on parameterized nodes.
+
+
+def f_target(x):
+    return np.pi * x[:, 0] + 1.0 + np.pi
```

Comment:
why the `+ 1.0` here?

Reply:
no reason, just to avoid the same constant twice. i've changed this to `math.pi * x[:, 0] + math.e` now to avoid such confusion